tooling/github-strategy.md

```diff
@@ -0,0 +1,92 @@
+
+The opportunity to balance both needs comes from observation that there is not much overlap between the repositories that each perspective cares about the most.
+
+## Visibility Differences by InnerSource Project Type
+
+There are several types of InnerSource projects, each with its own characteristics and considerations. The table provides an overview of these project types and their visibility differences within an organization. What kinds of InnerSource seeds does your organization have? What areas in your organization would you like to start with when strategically adopting InnerSource? Use this categorization to help you categorize your project.
+
+| Type of InnerSource​ | InnerSource Participation method​ | Significant Communication before pull request or reuse?​ | Will owning party likely know non-owning party beforehand​ | numbers of chances for InnerSource participation killed off by org-level visibility boundaries​ | Enterprise impact of InnerSource action not occurring​ | Likely sensitivity of the code if maliciously released​ |
+| --- | --- | --- | --- | --- | --- | --- |
+| Learning, copy and pasting (examples, templates)​ | Learning​ | None​ | No​ | Most​ | Entropy and lost time.​ | Probably low​ |
+| Use Reusable tools built for Company circumstances. ​ | Reuse​ | None-little​ | No​ | Many​ | Loss of efficiency and speed. ​ | Probably low​ |
+| Content change, fix, add, or update (websites, documentation)​ | Contribution, small ones​ | None-little​ | No​ | Many​ | Wasted time due to bad information.​ | Probably low​ |
+| Build within someone else's internal service | Contribution, within well-defined parameters​ | Yes​ | No​ | Some​ | More things never done at all.​ | Probably low but more exceptions.​ |
+| Deduplication. Don't built same thing twice, build general solution.​ | Contribution ​ | Yes​ | Varies​ | Some​ | General purpose code written less often. ​ | Depends, some very high.​ |
+| Ensure alignment between related projects​ | Contribution ​ | Yes​ | Likely​ | A few​ | Loss of efficiency and speed.​ | Depends, some very high.​ |
+| Not get slowed down by who is owning ​ | Contribution ​ | Yes​ | Likely​ | A few​ | Loss of efficiency and speed.​ | Depends, some very high.​ |
+
+## Project Participation Difficulty Level by Setting
+
+The difficulty level to join a project depends on your access rights to GitHub Enterprise and the repository type. It should be noted that GitHub settings in the enterprise often use SCIM and work with IdPs such as Microsoft Entra and Okta, but the participation hurdle can also vary depending on the settings on the IdP side.
+
+Here are groups that can be used to define repository permissions ordered by process toil to join:
+
+1. Everyone behind company firewall
+1. Internal visibility​ (everyone on platform is in this group)
+1. Security group created automatically​
+1. Security group with self-join and NO forced renew after X time period​
```

Comment:
Interesting options - never heard of them before. Is this also available on GHES or only on GHEC or GHC?

Reply:
This is not a GitHub Enterprise configuration item, but rather something that can be configured via IdPs like Microsoft Entra (Azure AD) or Okta, or via APIs in those configurations.
I mention it here because the group settings on the IdP side will be synced when you sync with SCIM, but obviously this context is out of the context of GitHub Enterprise settings, so it might be better to write it separately.

Reply:
There are many companies that use SCIM to coordinate their teams, so I think it would be a good idea to mention it somewhere.
However, if you integrate with SCIM, the identity management part of GitHub may not be in the hands of GitHub Admin but in the hands of Azure AD administrator. So, could SCIM integration be the anti-pattern of InnerSource?

The GitHub Admin might think more about Developer Experience and InnerSource, but the Azure AD Admin is an IT person in many cases, and might not think much about that context. This is just an assumption though